frontend/src/components/commons/PlaceForm.js

```diff
@@ -0,0 +1,363 @@
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import Grid from '@material-ui/core/Grid';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Tooltip from '@material-ui/core/Tooltip';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import CancelOutlinedIcon from '@material-ui/icons/CancelOutlined';
+import SaveOutlinedIcon from '@material-ui/icons/SaveOutlined';
+import MenuItem from '@material-ui/core/MenuItem';
+
+const useStyles = makeStyles((theme) => ({
+  form: {
+    width: '100%', // Fix IE 11 issue.
+    marginTop: theme.spacing(3),
+  },
+  button: {
+    margin: theme.spacing(3, 0, 2),
+  },
+}));
+
+const initialState = {
+  primaryPictureUrl: '',
+  type: 'cityscape',
+  title: '',
+  street: '',
+  address: '',
+  latitude: '',
+  longitude: '',
+  placeDescription: '',
+  pictureDescription: '',
+  aperture: '',
+  focalLength: '',
+  shutterSpeed: '',
+  iso: '',
+  flash: '',
+  youTubeUrl: '',
+  extraOne: '',
+  extraTwo: '',
+  particularities: '',
+};
+
+const types = [
+  {
+    value: 'cityscape',
+    label: 'Stadt',
+  },
+  {
+    value: 'harbour',
+    label: 'Hafen',
+  },
+  {
+    value: 'landscape',
+    label: 'Landschaft',
+  },
+  {
+    value: 'night',
+    label: 'Nachtaufnahme',
+  },
+];
+
+export default function PlaceForm({ onSave, place = initialState }) {
+  const classes = useStyles();
+  const history = useHistory();
+  const [placeData, setPlaceData] = useState(place);
+
+  return (
+    <>
+      {/*<UploadPicture />*/}
+      <form className={classes.form} noValidate onSubmit={handleSubmit}>
+        <Grid container spacing={2}>
+          <Grid item xs={12}>
+            <TextField
+              value={placeData.title}
+              onChange={handleChange}
+              name={'title'}
+              variant={'outlined'}
+              fullWidth
+              id={'title'}
+              label={'Titel des Bildes'}
+              required
+              autoFocus
+            />
```

Comment:
There are many TextFields here. `value`, `name` and `always` have the same value. I smell a new component here :)

Reply:
aaahhh